packages/pipeline/src/Pipeline.tsx

```diff
@@ -0,0 +1,285 @@
+ * Handles overflowed content gracefully by indicating how many other stages are hidden.
+ *
+ * ```
+ * <Pipeline>
+ *   <Stage>$match</Stage>
+ *   <Stage>$addFields</Stage>
+ *   <Stage>$limit</Stage>
+ * </Pipeline>
+ * ```
+ * @param props.children Content that will appear inside of the Pipeline component.
+ * @param props.className Classname applied to Pipeline content container.
+ * @param props.size Alters the rendered size of the component.
+ * @param props.variant Alters the visual apperance of the component.
+ */
+const Pipeline = forwardRef(
+  (
+    {
+      children,
+      className = '',
+      size = Size.XSmall,
+      variant = Variant.Default,
+      ...rest
+    }: PipelineProps,
+    ref: Ref<HTMLDivElement>,
+  ): ReactElement => {
+    // State
+    const [pipelineNode, setPipelineNode] = useElementNode();
+    const [hasHiddenStages, setHasHiddenStages] = useState(false);
+
+    // Handlers
+
+    /**
+     * Determines whether the Pipeline element is overflowed.
+     * If the Pipeline is overflowed, this means that we have stages which aren't visible and
+     * so we should display the counter.
+     */
+    const handleCounterDisplay = () => {
+      const result = isElementOverflowed(pipelineNode);
+      setHasHiddenStages(result);
+    };
+
+    /**
+     * Decorates the last visible stage with a className necessary for applying
+     * the correct visual appearance. The mutation of the DOM is required otherwise the
+     * Stage components will re-render, triggering an infinite loop on the
+     * mutation observer.
+     */
+    const setLastVisibleStage = () => {
+      const allStages = Array.from<HTMLElement>(pipelineNode.childNodes);
+
+      // Remove previously added last visible stage classNames
+      allStages.forEach(element => {
+        element.classList.remove(lastVisibleClassName);
+      });
+
+      // Get the last visible stage
+      const lastVisibleStage = findLast(
+        allStages,
+        element => element.dataset.stageVisible !== 'false',
+      );
+
+      // Decorate the last visible stage so that we can appropriately style the chevron
+      if (lastVisibleStage) {
+        lastVisibleStage.classList.add(lastVisibleClassName);
+      }
+    };
+
+    /**
+     * Callback for the Mutation Observer.
+     * @param records The records for the ovserved mutation
+     */
+    const observeChanges = (records: Array<MutationRecord>) => {
+      const types = records.map(r => r.type);
+      const attrs = records.map(r => r.attributeName);
+
+      if (attrs.includes('data-stage-visible') || types.includes('childList')) {
+        handleCounterDisplay();
+        setLastVisibleStage();
+      }
+    };
+
+    // Effects
+    useMutationObserver(
+      pipelineNode,
+      {
+        childList: true,
+        subtree: true,
+        attributes: true,
+      },
+      observeChanges,
+    );
+
+    const _children = React.Children.map(children, child => {
```

Comment:
Is there a more semantic name for this variable?

Reply:
Resolved by ffa74d1